### ISSUE TITLE
Add `explore --try`

### ISSUE THREAD

**Issue by texastoland** [nushell/nushell#12116]
### Related problem

I pretty much always want `:try`.

### Describe the solution you'd like

Launch in that mode.

**Comment by fdncred** [comment]
That's interesting. I don't think we've had that request before. Can you explain your workflow and what you're doing with `:try`? I'm just curious.

**Comment by texastoland** [comment]
Yeah! I **freaking** love `explore`. I'm literally trying it in my `display_output` hook today. To me it's like [Broot](https://github.com/Canop/broot) (exa+rg+bat-in-1) for data. I like to start navigating a structure like `$env` or `sys` or `glob` then starting manipulating it (mostly `where` but occasionally other filters) like a REPL. There are other features I'd like but this flag is just the 1 I feel confident in implementing myself 😸

**Comment by texastoland** [comment]
Just started on this. Making a short list of other things to try:

1. Configurable shortcuts
2. Row navigation
3. Simplified navigation in general
4. Fuzzy filtering (in contrast to finding)

1 PR or 1 per feature?

**Comment by fdncred** [comment]
1 per feature please

**Comment by hustcer** [comment]
I'm wondering if we can display data in Explorer and refresh the data every a few seconds?

**Comment by texastoland** [comment]
> I'm wondering if we can display data in Explorer and refresh the data every a few seconds?

Interesting idea but out of scope for this. I also suspect streaming would be a better approach than polling based on @devyn's recent work.

**Comment by devyn** [comment]
@texastoland 

> Like `tail`! Interesting but out of scope for this. I also suspect streaming would be a better approach than polling based on @devyn's recent work.

Streams were already there, I just added them to plugins :)

It would be good to have `explore` be able to read a stream and gradually update the output. I don't think it can do that right now

**Comment by texastoland** [comment]
@hustcer Added to the list ✅

**Comment by sbrow** [comment]
I would like to add `yank` functionality to this list. Maybe `y` to copy the selected column into the clipboard or `Y` to copy the entire row?

I often find myself scrolling through a table and am surprised when I have to pull out the mouse to copy the column data. :)

**Comment by texastoland** [comment]
> I would like to add `yank` functionality to this list. Maybe `y` to copy the selected column into the clipboard or `Y` to copy the entire row?

Good idea if the clipboard plugin is installed ✅

**Comment by fdncred** [comment]
We're also trying to move away from vim-like behavior. That doesn't mean we can't have definable keyboard shortcuts that do certain things, but it does mean that we'd like to eliminate modes altogether from explore.

**Comment by texastoland** [comment]
Yeah I put it low on the list because I know it need discussed. My idea was basic shortcuts inspired by broot, Lazygit, Micro, etc. I'll raise anything in #design-discussion before attempting it.

**Comment by texastoland** [comment]
Tracked https://github.com/okbob/pspg from Discord as our model for navigation 🔥

**Comment by fdncred** [comment]
pspg is nice but we need to be able to drill down into nested data. We could do that by letting the user hit enter on a cell to drill into or perhaps having a tui tree widget.

**Comment by albert-tomanek** [comment]
> To me it's like https://github.com/Canop/broot (exa+rg+bat-in-1) for data.

@texastoland I had just the same idea today. I actually opened an issue with a suggestion for what broot-style usage of `explore` could look like at #12221

**Comment by texastoland** [comment]
Broot is hands down my favorite CLI. I totally envision `explore` as a Broot-for-data-and-files-too! I have less confidence in my attention span though so keeping notes 🙊

**Comment by albert-tomanek** [comment]
> Auto-navigating values:
> closures - explain
> dirs - ls
> files - open
> urls - http get
> modules/commands - viwe source

I think one design decision we need to make is how much functionality we want baked in to `explore`. Because I see two different options: either the handling of urls, directory contents, etc. can be implemented in Rust code (making it a kind of universal everything-browser), or `explore` could just be a data _display_ tool, and the data would be obtained from user-provided closures (as in #12221 ). I would personally be more partial to the second option because I think this would make it applicable in a more versatile range of uses: the same structure (and browsing needs) of data can arise from two completely different data sources (eg. YT comments and a `ls` tree). If we went down this route, a default wrapper fn could be provided for `explore` that would invoke it with closures for handling links/directories/etc. so that the user wouldn't have to type them every time.  

I suppose this is almost a MIME-type style scenario where there is an unlimited amount of data types

**Comment by texastoland** [comment]
I'm strongly on the side of it "just works". I suppose there could be a closure flag to transform data before reaching it?

**Comment by fdncred** [comment]
We're fine with adding a tree-widget and the tooling to hand that because Sophia and I have discussed that before, but we want the main functionality to be paging and data exploring. We've already removed some functionality that kind of didn't belong. So, we don't want to make a bulky app/command that does "everything". It's hard to talk in generalities so we'll need to discuss each feature on a case by case basis.

**Comment by albert-tomanek** [comment]
@texastoland  I think our two visions are compatible, what I had in mind was that we could do something like:
```rb
def browse {   # Official nushell command
    explore --handle-url {|| curl $in ... } --handle-file {|| xdg-open $in} --children {|| ls $in }
}

# Custom, user-defined behavior
> yt comments "http://..." | explore --children {|comment| $comment.replies}
```
The 'just works' functionality would be achieved by using the wrapper (your use case), whilst the UI & filtering funcitonality would be kept modular so that it could still be used for browsing custom data like scraped YT comments (my use case)

**Comment by HRKings** [comment]
May I suggest a mode that works something like: https://github.com/ynqa/empiriqa ? I would love the instant feedback and the ability to "disable" parts of the pipeline I'm exploring

**Comment by fdncred** [comment]
Looks interesting. BTW, we have "instant" feedback already in `:try` mode. You may need this configuration too.
```nushell
  explore: {
    config: {
        cursor_color: 'red'
    }
    table: {
        selected_cell: { bg: 'blue'}
        show_cursor: false
    }
    try: {
        reactive: true
    }
  }
```
